File: php/features.php
<?php
declare(strict_types=1);

// Ceorater SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CeoraterFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CeoraterBaseFeature();
            case "test":
                return new CeoraterTestFeature();
            default:
                return new CeoraterBaseFeature();
        }
    }
}

File: py/features.py
# Ceorater SDK feature factory

from feature.base_feature import CeoraterBaseFeature
from feature.test_feature import CeoraterTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CeoraterBaseFeature(),
        "test": lambda: CeoraterTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
